pkg/registry/cachesize/cachesize.go

```diff
@@ -36,6 +36,7 @@ func NewHeuristicWatchCacheSizes(expectedRAMCapacityMB int) map[schema.GroupReso
 	watchCacheSizes := make(map[schema.GroupResource]int)
 	watchCacheSizes[schema.GroupResource{Resource: "replicationcontrollers"}] = maxInt(5*clusterSize, 100)
 	watchCacheSizes[schema.GroupResource{Resource: "endpoints"}] = maxInt(10*clusterSize, 1000)
+	watchCacheSizes[schema.GroupResource{Resource: "endpointslices", Group: "discovery.k8s.io"}] = maxInt(10*clusterSize, 1000)
```

Comment:
are these 1:1 with endpoints? I thought there were expected to be fewer of these. cc @robscott

Reply:
They are not, it's a one Endpoints object to many EndpointsSlice objects relation.
But what matters for the cache size is the number of updates which is roughly the same, i.e. for a single pod change there will be one Endpoints object update and also one EndpointSlice object update.

Reply:
Yes - this cache is not number of objects, but number of recent changes (events) we keep.
Roughly 1:1 correspondence here is what we want.

Reply:
It seems problematic that this is fixed and open-coded.  Even *I* did not know to change this.

What can we do to make this automatic and heuristic rather than manual?

Reply:
We should - I remember having an issue for it, but can't find it now. But yeah - it's in our backlog.